core/geotime/src/main/java/mil/nga/giat/geowave/core/geotime/store/filter/SpatialQueryFilter.java

```diff
@@ -196,8 +204,73 @@ public void fromBinary(
 					fieldId));
 		}
 		buf.get(theRest);
-		queryGeometry = geometryInterner.intern(GeometryUtils.geometryFromBinary(geometryBinary));
+		// cannot intern the prepared geometry. It uses a Object.hashCode().
+		preparedGeometryImage = geometryImageInterner.intern(new GeometryImage(
+				geometryBinary));
+		preparedGeometryImage.init();
 
 		super.fromBinary(theRest);
 	}
+
+	/**
+	 * Used for interning Geometry. Interning a geometry provides no additional
+	 * benefit since the geometry is already constructed in the heap at the
+	 * interning begins. The byte image of geometry provides a more efficient
```

Comment:
A geometry is constructed first prior to interning.  It consumes heap space before being released to its interned duplicate. This approach prevents a Geometry from being constructed if it already exists.